Source/Redemption/Private/RedemptionGameInstance.cpp

```diff
@@ -1,19 +1,26 @@
 // Fill out your copyright notice in the Description page of Project Settings.
 
-
 #include "RedemptionGameInstance.h"
+#include "RedemptionUtils.h"
 
-FVector URedemptionGameInstance::GetPlayerDiedLocation()
+URedemptionGameInstance::URedemptionGameInstance()
 {
-	
-	return PlayerDiedLocation;
-	
+	for (int i = 0; i < NUM_LEVELS; i++)
+	{
+		PlayerDiedLocation.Add(FVector::ZeroVector);
+		CompletedLevels.Add(false);
+	}
 }
 
-void URedemptionGameInstance::SetPlayerDiedLocation()
+FVector URedemptionGameInstance::GetPlayerDiedLocation(int LevelId)
+{
+	return PlayerDiedLocation[LevelId];
+}
+
+void URedemptionGameInstance::SetPlayerDiedLocation(int LevelId)
 {
 	ACharacter* myCharacter = UGameplayStatics::GetPlayerCharacter(GetWorld(), 0);
-	PlayerDiedLocation = myCharacter->GetActorLocation();
+	PlayerDiedLocation[LevelId] = myCharacter->GetActorLocation();
```

Comment:
I feel like you should clear out the other level died locations when you set a new one. This lead to some bugs in playthroughs where a player would leave the tutorial, go into level one then next time the next player went into the tutorial it had a body there. Maybe do a loop in here to clear them all first then set the current one?

Reply:
It is intended to the player to still have it's body there even if they go to another level